radicle/src/cob/identity.rs

```diff
@@ -0,0 +1,680 @@
+use std::{ops::Deref, str::FromStr};
+
+use crypto::{PublicKey, Signature};
+use once_cell::sync::Lazy;
+use radicle_cob::{ObjectId, TypeName};
+use radicle_crdt::{clock, GMap, Gate, LWWMap, LWWReg, Max, Redactable, Semilattice};
+use radicle_crypto::{Signer, Verified};
+use radicle_git_ext::Oid;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::{
+    cob::{
+        self,
+        common::Timestamp,
+        store::{self, FromHistory as _, Transaction},
+    },
+    identity::{doc::DocError, Identity},
+    prelude::Doc,
+    storage::{git as storage, RemoteId},
+};
+
+use super::{
+    thread::{self, Thread},
+    Author, OpId,
+};
+
+/// The logical clock we use to order operations to patches.
+pub use clock::Lamport as Clock;
+
+/// Type name of a patch.
+pub static TYPENAME: Lazy<TypeName> =
+    Lazy::new(|| FromStr::from_str("xyz.radicle.identity.proposal").expect("type name is valid"));
```

Comment:
```suggestion
    Lazy::new(|| FromStr::from_str("xyz.radicle.id.proposal").expect("type name is valid"));
```
More familiar, since we use `rad/id` ref and `rad id` sub-command?